skimage/filters/thresholding.py

```diff
@@ -174,6 +175,7 @@ def threshold_yen(image, nbins=256):
     >>> thresh = threshold_yen(image)
     >>> binary = image <= thresh
     """
+    assert_nD(image, 2)
```

Comment:
I'm fairly sure most of these non-adaptive algorithms act globally, and thus don't have any constraints on the rank of the input.

We shouldn't impose a 2d restriction.